src/period/index.js

```diff
@@ -19,8 +24,14 @@ module.exports = (bridgeState, nodeConfig = {}) => async (rsp, chainInfo, chainH
 
   logPeriod('checkBridge');
   // use a timeout to relax race conditions (if period submission is fast)
-  await new Promise((resolve) => {
+  await new Promise(resolve => {
     setTimeout(async () => {
+      await submitPeriodVote(
```

Comment:
why did you decide to call `submitPeriodVote()` here?

this is called when tendermint is trying to propose a new block (right?), and `submitPeriod()` is called as a fallback, in case the selected proposer is offline.

Reply:
this handler is called every period (`checkPeriod` middleware). Right, I'm submitting the vote here as a fallback as well (`submitPeriodVote` has a protection against double votes).

`updatePeriod` is being called every block and that's where I submit a vote first: https://github.com/leapdao/leap-node/pull/305/files/15924f654aabe89a4492c03fa3c51f7f67b02545#diff-786c9ee7cb87a989994653f4a23150a6R25